esp32c3-hal/examples/README-embassy.md

```diff
@@ -0,0 +1,27 @@
+
+## Embassy Examples
+
+* 😁 Decision Point1: choose systick vs timg0 
+- systick is for ARM cortex-m, the esp-hal provides a simple 24-bit clear on write decrementing wrap-on-zero counter for os-scheduling sed for basic countdown timer, interrpt generation on reaching zero. Better for ARM, etc. compatibility.
```

Comment:
Where did you get this from?

I'd suggest familiarizing yourself with the ESP32 lineup and their peripherals. I don't think any ARM-based ESP32 exists, the C3 certainly isn't ARM. The SysTimer peripheral, that this feature coopts as the timebase is a 52 bit timer peripheral present in most ESP32s (it's there at least in the S2, S3, C3, and maybe more).

Reply:
@bugadani - see cunninghams law. 

I'm attempting to sift through the code and understand it, write up documentation with my guesses and hopefully get them corrected either in github or matrix.  

I assumed (incorrectly) the systick was put in to provide compatibility for embassy-rs which seems stm32 centric, that's just what I was able to find/infer from searching & asking chatgpt.  the documentation in this area is sparse. 

I was trying to say that systick might be a better choice than timg0 if multiple architectures are important - because apparently I must choose one, and only one timer with no explanation why one is better or different than the other.